### PR TITLE
Fixed and tested changes, adding and removing

### PR DESCRIPTION
**Pull request by Lobotuerk**
Fixed @hidmic comments, fixed changing, adding and removing. TODO: change the used_priorities_ to local instead of global as Mich wanted to, just want to PR the working version first

### CODE REVIEW

File: src/cmd_vel_mux.cpp [hunk 1]
```diff
@@ -113,7 +114,7 @@ bool CmdVelMux::parametersAreValid(const std::map<std::string, ParameterValues>
       RCLCPP_WARN(get_logger(), "Empty topic for '%s'", parameter.first.c_str());
       return false;
     }
-    else if (parameter.second.timeout < 0.0)
+    if (parameter.second.timeout < 0.0)
```
Comment (hidmic) [inline src/cmd_vel_mux.cpp:117]: @Lobotuerk you may remove all `else` clauses here.

File: src/cmd_vel_mux.cpp [hunk 2]
```diff
@@ -135,15 +136,14 @@ bool CmdVelMux::parametersAreValid(const std::map<std::string, ParameterValues>
...
 void CmdVelMux::configureFromParameters(const std::map<std::string, ParameterValues> & parameters)
 {
-  std::map<std::string, ParameterValues>::const_iterator parameter;
   std::map<std::string, std::shared_ptr<CmdVelSub>> new_map;
 
   for (const std::pair<std::string, ParameterValues> & parameter : parameters)
   {
-    std::string key = parameter.first;
-    ParameterValues parameter_values = parameter.second;
+    const std::string &key = parameter.first;
```
Comment (hidmic) [inline src/cmd_vel_mux.cpp]: @Lobotuerk nit:

```suggestion
    const std::string & key = parameter.first;
```

Same elsewhere.

File: src/cmd_vel_mux.cpp [hunk 3]
```diff
@@ -185,7 +181,7 @@ void CmdVelMux::configureFromParameters(const std::map<std::string, ParameterVal
   double longest_timeout = 0.0;
   for (std::pair<const std::string, std::shared_ptr<CmdVelSub>> & m : map_)
   {
-    std::string key = m.first;
+    const std::string &key = m.first;
```
Comment (hidmic) [inline src/cmd_vel_mux.cpp]: @Lobotuerk nit: consider doing the same aliasing for `m.second`.

File: src/cmd_vel_mux.cpp [hunk 4]
```diff
@@ -223,6 +219,28 @@ void CmdVelMux::configureFromParameters(const std::map<std::string, ParameterVal
...
+    if (allowed_ == input_name)
+    {
+      // Take down the deleted subscriber if it was the one being used as source
+      allowed_ = VACANT;
+
+      // ...notify the world that nobody is publishing on cmd_vel; its vacant
+      auto active_msg = std::make_unique<std_msgs::msg::String>();
+      active_msg->data = "idle";
+      active_subscriber_pub_->publish(std::move(active_msg));
+    }
```
Comment (hidmic) [inline src/cmd_vel_mux.cpp]: @Lobotuerk `addInputToParameterMap` should not have an effect on the current state. Only `configureParameters` can carry out this operation.

**Review by hidmic — CHANGES_REQUESTED** [review]

**Comment by Lobotuerk** [comment]
@hidmic

**Review by hidmic — APPROVED** [review]
Way to go man, LGTM!

**Review by clalancette — APPROVED** [review]
Looks good!  I'm going to merge this into my PR, but I won't merge the whole thing to master until next week.  Thanks @Lobotuerk !

**Pull request merged**